source/opt/combine_access_chains.cpp

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "combine_access_chains.h"
+
+#include "constants.h"
+#include "ir_builder.h"
+#include "ir_context.h"
+
+namespace spvtools {
+namespace opt {
+
+Pass::Status CombineAccessChains::Process() {
+  bool modified = false;
+
+  for (auto& function : *get_module()) {
+    modified |= ProcessFunction(function);
+  }
+
+  return (modified ? Status::SuccessWithChange : Status::SuccessWithoutChange);
+}
+
+bool CombineAccessChains::ProcessFunction(Function& function) {
+  bool modified = false;
+
+  cfg()->ForEachBlockInReversePostOrder(
+      function.entry().get(), [&modified, this](BasicBlock* block) {
+        block->ForEachInst([&modified, this](Instruction* inst) {
+          switch (inst->opcode()) {
+            case SpvOpAccessChain:
+            case SpvOpInBoundsAccessChain:
+            case SpvOpPtrAccessChain:
```

Comment:
Also OpInBoundsPtrAccessChain  ?

Reply:
I've added support, but not pushed yet. Will come with the rebase.

Reply:
Resolved.